icon/chainscore_iiss.go

```diff
@@ -140,15 +140,20 @@ func (s *chainScore) Ex_setStake(value *common.HexInt) (err error) {
 	prevTotalStake := ia.GetTotalStake()
 
 	//update IISS account
+	revision := s.cc.Revision().Value()
+	expireHeight := s.cc.BlockHeight() + calcUnstakeLockPeriod(es.State, tStake, tSupply).Int64()
 	var tl []icstate.TimerJobInfo
 	switch stakeInc.Sign() {
 	case 1:
 		// Condition: stakeInc > 0
-		tl, err = ia.DecreaseUnstake(stakeInc)
+		tl, err = ia.DecreaseUnstake(stakeInc, expireHeight, revision)
 	case -1:
-		expireHeight := s.cc.BlockHeight() + calcUnstakeLockPeriod(es.State, tStake, tSupply).Int64()
 		slotMax := int(es.State.GetUnstakeSlotMax())
```

Comment:
add revision parameter

Reply:
Discard this suggestion.

chainscore.handleRevisionChange() sets unstakeSlotMax to an appropriate value.